javascript/javascript2/week3/home-work.js

```diff
@@ -0,0 +1,116 @@
+// Log out the text Called after 2.5 seconds 2.5 seconds after the script is loaded.
+// Create a function that takes 2 parameters: delay and stringToLog.Calling this function
+//     should log out the stringToLog after delay seconds.C
+// all the function you have created with some different arguments.
+// Create a button in html.When clicking this button, use the function you
+// created in the previous task to log out the text: Called after
+//  5 seconds 5 seconds after the button is clicked.
+function delayMsg(delay, stringToLog) {
+    setTimeout(() => {
+        console.log(`Called after 5 seconds`);
+    }, delay * 1000)
+}
+const printMSg = delayMsg;
+const button1 = document.getElementById("1st");
+button1.addEventListener('click', callBack => {
+    printMSg(5, "Called after 5 seconds");
+});
+
+// Create two functions and assign them to two different variables.One function logs out Earth,
+//  the other function logs out Saturn.Now create a new third function that has one parameter:
+//   planetLogFunction.The only thing the third function should do is call the provided parameter
+//   function.
+// Try call the third function once with the Earth function and once with the Saturn function.
+const elogger = () => console.log(`earth`);
+const slogger = () => console.log(`saturn`);
+elogger();
+slogger();
+
+const planetLogFunction = (fun) => fun();
```

Comment:
You could replace the "fun" name since it can be hard to understand for other developers what kind of function should be used as the parameter. Another approach could be:
```
const earthLogger = () => {
  console.log("Earth")
}
const saturnLogger = () => {
  console.log("Saturn")
}

const planetLogger = (loggingFunction) => {
  loggingFunction()
}

planetLogger(earthLogger)
```

Reply:
👍🏻